### ISSUE TITLE
`isinstance` narrowing ignores covariance of `typing.Type`?

### ISSUE THREAD

**Issue by parched** [microsoft/pyright#3607]
We have some code that does an `isinstance` check on a dynamic tuple of types, but afterwards the type is inferred as `Never`.
It looks like `type.Type` is being treated as invariant instead of covariant.

```python
SOME_TYPES_L: List[Type[object]] = [int, float]
SOME_TYPES: Tuple[Type[object], ...] = tuple(SOME_TYPES_L)

def check_object(obj: object):
    if isinstance(obj, SOME_TYPES):
        return
    elif isinstance(obj, list):
        for i in obj:  # "Never" is not iterable
            pass
```

I expected `obj` to be narrowed to `list` under the `elif`.

**VS Code extension or command-line**
Pylance language server 2022.6.20 (pyright 979d9c9e)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.256, which I just published. It will also be included in the next release of pylance.
